src/attributes/attributes.h

```diff
@@ -26,15 +26,54 @@
 extern "C" {
 #endif
 
+#include <stdio.h>
+#include <inttypes.h>
+
 #include <common.h>
 
 /**
- * Output object attributes.
- * @param session Valid PKCS11 session.
- * @param object The object handle.
- * @return CK_RV Value returned by the PKCS#11 library. This will indicate success or failure.
+ * get single object attribute
+ *
+ * @returns CK_RV Value returned by the PKCS#11 library. This will indicate
+ *   success or failure.
+ */
+CK_RV attributes_get(
+        /** [in] valid PKCS11 session */
```

Comment:
I think this is level of documentation we should shoot for (and our other samples are not here yet).